### PR TITLE
Wrap device descriptor generation and retrieval in a method.

### LINKED ISSUE #1882: Some `AliasTest`s run into segmentation fault with NVFUSER_PROF=print.
```
$ git rev-parse HEAD
c086baba6d780600184b90b90c54260c5d83095d
```

```
$ NVFUSER_PROF=print bin/nvfuser_tests --gtest_filter=AliasTest.View
Running main() from /opt/pytorch/nvfuser/third_party/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = AliasTest.View
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from AliasTest
[ RUN      ] AliasTest.View
[nvdl-a112-d003:35338:0:35338] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x30)
==== backtrace (tid:  35338) ====
 0 0x0000000000042520 __sigaction()  ???:0
 1 0x000000000078cd07 nvfuser::FusionProfiler::stop()  ???:0
 2 0x00000000004c6c8d nvfuser::FusionExecutorCache::runFusionWithInputs()  ???:0
 3 0x0000000000809811 nvfuser::AliasTest_View_Test::TestBody()  ???:0
 4 0x0000000000ce4351 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  :0
 5 0x0000000000cd0f95 testing::Test::Run()  gtest-all.cc:0
 6 0x0000000000cd1722 testing::TestInfo::Run()  ???:0
 7 0x0000000000cd22ab testing::TestSuite::Run()  gtest-all.cc:0
 8 0x0000000000cda904 testing::internal::UnitTestImpl::RunAllTests()  ???:0
 9 0x0000000000cd1905 testing::UnitTest::Run()  ???:0
10 0x0000000000297664 main()  ???:0
11 0x0000000000029d90 __libc_init_first()  ???:0
12 0x0000000000029e40 __libc_start_main()  ???:0
13 0x0000000000297c75 _start()  ???:0
=================================
[1]    35338 segmentation fault (core dumped)  NVFUSER_PROF=print bin/nvfuser_tests --gtest_filter=AliasTest.View
```

Works with NVFUSER_PROF=print.nocupti though. 

```
$ NVFUSER_PROF=print.nocupti bin/nvfuser_tests --gtest_filter=AliasTest.View
Running main() from /opt/pytorch/nvfuser/third_party/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = AliasTest.View
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from AliasTest
[ RUN      ] AliasTest.View

Fus#  NSegs CuEvtTm(ms) HstTm(ms) CmpTm(ms)
    0     1      54.117    54.107    53.518
[       OK ] AliasTest.View (156 ms)
[----------] 1 test from AliasTest (156 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (156 ms total)
[  PASSED  ] 1 test.
```

### PR DESCRIPTION
**Pull request by wujingyue**
So it's harder to make mistakes not checking existence.

Currently, when no nvfuser kernels are launched (e.g. when the fusion computes merely aliases), https://github.com/NVIDIA/Fuser/blob/6c066c0c1f01ae984bdb6adf51f5edc4c529c023/csrc/fusion_profiler.cpp#L726-L736 is never executed and therefore https://github.com/NVIDIA/Fuser/blob/6c066c0c1f01ae984bdb6adf51f5edc4c529c023/csrc/fusion_profiler.cpp#L787 reads `fp->device_descriptors_[0]` while it's empty. 

With this PR, all reads from `fp->device_descriptors_` are through the `fp->deviceDescriptor` method, which if needed resizes the array and generates device info. 

Fixes #1882.

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

**Review by naoyam — COMMENTED** [review]
Looks good but should be stamped by @kevinstephano

**Comment by wujingyue** [comment]
!build

File: csrc/fusion_profiler.cpp [hunk 1]
```diff
@@ -669,7 +669,7 @@ SegmentProfiler& FusionProfiler::segment(size_t idx) {
   return get()->segments_.at(idx);
 }
 
-void FusionProfiler::start(bool cupti_disable) {
+/*static*/ void FusionProfiler::start(bool cupti_disable) {
```
Comment (kevinstephano) [inline csrc/fusion_profiler.cpp:672]: Did you mean to leave the commented static?

File: csrc/fusion_profiler.cpp [hunk 2]
```diff
@@ -686,7 +686,22 @@ void FusionProfiler::start(bool cupti_disable) {
...
+
+  if (desc.device != device_id) {
+    // This happens when device_descriptors_[device_id] is initialized (and
+    // thus device==-1) but not populated.
+    DeviceDescriptor::generate(desc, device_id);
+  }
+  return desc;
+}
+
+/*static*/ void FusionProfiler::stop() {
```
Comment (kevinstephano) [inline csrc/fusion_profiler.cpp:704]: Same comment here about static.

**Review by kevinstephano — APPROVED** [review]
Looks good, just added a couple nit comments.

File: csrc/fusion_profiler.cpp (hunk 1 above)
Comment (wujingyue) [reply csrc/fusion_profiler.cpp:672]: Yes. I tend to do this extensively to remind readers this is a static method.

**Pull request merged**